src/carousel/Carousel.js

```diff
@@ -1297,41 +1066,58 @@ export default class Carousel extends Component {
                 // and https://github.com/facebook/react-native/issues/13100#issuecomment-328986423
                 { width: sliderWidth, flexDirection: this._needsRTLAdaptations() ? 'row-reverse' : 'row' }
         ];
+
+        const innerMarginStyle = vertical ? {
+            paddingTop: this._getContainerInnerMargin(),
+            paddingBottom: this._getContainerInnerMargin(true)
+        } : {
+            paddingLeft: this._getContainerInnerMargin(),
+            paddingRight: this._getContainerInnerMargin(true)
+        };
+
         const contentContainerStyle = [
-            vertical ? {
-                paddingTop: this._getContainerInnerMargin(),
-                paddingBottom: this._getContainerInnerMargin(true)
-            } : {
-                paddingLeft: this._getContainerInnerMargin(),
-                paddingRight: this._getContainerInnerMargin(true)
-            },
-            contentContainerCustomStyle || {}
+            contentContainerCustomStyle || {},
```

Comment:
Why does this come before the next line? I would like the ability to add extra padding if I have a Header / TabBar. The contentContainerCustomStyle padding gets overwritten if I choose not to use useExperimentalSnap

Reply:
I totally understand the need, but the issue is that it will then mess with the inner logic (the active item's calculation will be completely off). However, now that I think about it, with the updated `_getActiveItem()` logic it might just work out-of-the-box.

Do you mind doing a few tests on your end? If you move this line after the override line `1080` and change the padding, is `onSnapToItem` still firing reliably and with the proper index? With all 3 different `activeSlideAlignment` values?

Reply:
@AAAstorga Did you ever have a chance to test what we mentioned above? :-)